presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -2719,14 +2720,14 @@ private static void validateColumns(ConnectorTableMetadata tableMetadata)
         }
     }
 
-    // temporary, until variable precision timestamps are supported on write
-    private static void validateTimestampColumns(List<ColumnMetadata> columns)
+    // TODO validate timestamps in structural types (https://github.com/prestosql/presto/issues/5195)
+    private static void validateTimestampColumns(List<ColumnMetadata> columns, HiveTimestampPrecision timestampPrecision)
     {
         for (ColumnMetadata column : columns) {
             Type type = column.getType();
             if (type instanceof TimestampType) {
-                if (type != TIMESTAMP_MILLIS) {
-                    throw new PrestoException(NOT_SUPPORTED, "CREATE TABLE, INSERT and ANALYZE are not supported with requested timestamp precision: " + type);
+                if (((TimestampType) type).getPrecision() != timestampPrecision.getPrecision()) {
+                    throw new PrestoException(NOT_SUPPORTED, "Incorrect timestamp precision for " + type + "; the configured precision is: " + timestampPrecision);
```

Comment:
Use `format` for readability

Also, remove `:` since saying `is 123` reads more naturally than `is: 123`